src/main/kotlin/uk/gov/justice/digital/hmpps/assessrisksandneeds/services/RiskPredictorService.kt

```
@@ -59,11 +61,32 @@ class RiskPredictorService(
   }
 
   fun getAllRiskScores(identifierType: IdentifierType, identifierValue: String): List<AllPredictorVersioned<Any>> {
-    log.debug("Entered getAllRiskScoresVersioned for {}: {}", identifierType, identifierValue)
+    log.debug("Entered getAllRiskScores for ${identifierType.value}: $identifierValue")
     auditService.sendEvent(EventType.ACCESSED_RISK_PREDICTORS, mapOf(identifierType.value to identifierValue))
     communityClient.verifyUserAccess(identifierValue, RequestData.getUserName())
     val oasysRiskPredictorsDto = oasysClient.getRiskPredictorsForCompletedAssessments(identifierValue)
-    return AllPredictorVersionedLegacyDto.from(oasysRiskPredictorsDto)
+    return oasysRiskPredictorsDto
+      ?.assessments
+      ?.filter { it.assessmentType in listOf("LAYER3", "LAYER1") }
+      ?.map(AllPredictorVersionedLegacyDto::from)
+      .orEmpty()
+  }
+
+  fun getAllRiskScoresByAssessmentId(id: Long): AllPredictorVersioned<Any> {
+    log.debug("Entered getAllRiskScoresByAssessmentId for ID: $id")
+    auditService.sendEvent(EventType.ACCESSED_RISK_PREDICTORS_BY_ASSESSMENT_ID, mapOf("id" to id))
+    val oasysRiskPredictorsDto = oasysClient.getRiskPredictorsByAssessmentId(id)
+    return oasysRiskPredictorsDto
+      ?.assessments
+      ?.first()
+      ?.let { assessment: RiskPredictorAssessmentDto ->
+        val version = assessment.rsrScoreDto.rsrAlgorithmVersion?.toIntOrNull() ?: 0
```

Was concerned with this logic and whether it would be worth throwing an exception if the RSR algorithm version couldnt be found instead of assuming the assessment is in the legacy format.

---

based on the meeting with Howard earlier this logic will need to move away from the algorithm version. need to look at the docs tomorrow and confirm